### PR TITLE
chore: upgrade deps and tools

### PR DESCRIPTION
**Pull request by peterschutt**
- Addresses violations of new "safe-super" mypy check
- Addresses rich-click deprecation for importing `patch` from `rich_click.cli`
- Fixes test failure caused by `isinstance(command, Group)` no longer holding true after update to rich-click 1.8

## Description

-

### CODE REVIEW

File: litestar/cli/_utils.py [hunk 1]
```diff
@@ -252,7 +252,7 @@ def wrapped(ctx: Context, /, *args: P.args, **kwargs: P.kwargs) -> T:
 
 def _wrap_commands(commands: Iterable[Command]) -> None:
     for command in commands:
-        if isinstance(command, Group):
```
Comment (peterschutt) [inline litestar/cli/_utils.py:255]: This test failed after upgrading rich-click: https://github.com/litestar-org/litestar/blob/56b7395fa75a304dfea5711fe62646c469ce8616/tests/unit/test_cli/test_cli.py#L68

Issue is that, even though we import `Group` from `click`, it is actually `rich_click.patch._PatchedRichGroup`, and where `command` is an instance of `rich_click.rich_command.RichGroup`, `isinstance(command, Group)` returns `False`, meaning that the inner commands of groups weren't being wrapped to inject args.

This change seems to work, but I'm not familiar enough with click to know if there is a better way. @litestar-org/maintainers

Comment (provinzkraut) [reply litestar/cli/_utils.py:255]: Seems like an okay fix 👍

File: litestar/contrib/minijinja.py [hunk 2]
```diff
@@ -105,7 +105,6 @@ def __init__(self, directory: Path | list[Path] | None = None, engine_instance:
             directory: Direct path or list of directory paths from which to serve templates.
             engine_instance: A Minijinja Environment instance.
         """
-        super().__init__(directory, engine_instance)
```
Comment (provinzkraut) [inline litestar/contrib/minijinja.py:108]: Why are these removed? Iiuc this could cause issues when using these classes in multiple inheritance

**Review by provinzkraut — APPROVED** [review]
Approved with a question

File: litestar/contrib/minijinja.py (hunk 2 above)
Comment (peterschutt) [reply litestar/contrib/minijinja.py:108]: Mypy says it's unsafe: https://mypy.readthedocs.io/en/stable/error_code_list.html#check-that-call-to-an-abstract-method-via-super-is-valid-safe-super

> Iiuc this could cause issues when using these classes in multiple inheritance

Can you demo this for me please?

Comment (peterschutt) [reply litestar/contrib/minijinja.py:108]: Wouldn't the protocol `__init__` methods need to call super for that scenario too?

Comment (Alc-Alc) [reply litestar/contrib/minijinja.py:108]: I *think* these only happen when they are used as "mixins" in which case I would assume the protocol has to delegate to the super  `__init__`, we can maybe mark is as `final` indicating this should not be inherited?

Comment (peterschutt) [reply litestar/contrib/minijinja.py:108]: Shouldn't `MiniJinjaTemplateEngine` inherit from the protocol in this scenario?

Comment (Alc-Alc) [reply litestar/contrib/minijinja.py:108]: thank you, edited, so

>  the protocol __init__ methods need to call super for that scenario too?
seems like it should

Comment (provinzkraut) [reply litestar/contrib/minijinja.py:108]: > Wouldn't the protocol `__init__` methods need to call super for that scenario too?

Ah yeah, good point. Uh. I guess the question is how multiple inheritance works with protocols? Actually I'm not sure how they behave at runtime.

Comment (provinzkraut) [reply litestar/contrib/minijinja.py:108]: Okay, I checked. They don't seem to call `super().__init__`, so we'd have to add that explicitly to make them work with multi-inheritance anyway. In that case I'm fine with this change as-is.

**Review by JacobCoffee — APPROVED** [review]

**Comment by JacobCoffee** [comment]
> safe-super
